sentry/src/main/java/io/sentry/protocol/SentryTransaction.java

```diff
@@ -90,15 +91,18 @@ public SentryTransaction(final @NotNull SentryTracer sentryTracer) {
             tracerContext.getDescription(),
             tracerContext.getSamplingDecision(),
             tracerContext.getStatus(),
-            tracerContext.getOrigin()));
+            tracerContext.getOrigin());
+
+    contexts.setTrace(tracerContextToSend);
     for (final Map.Entry<String, String> tag : tracerContext.getTags().entrySet()) {
       this.setTag(tag.getKey(), tag.getValue());
     }
 
-    final Map<String, Object> data = sentryTracer.getData();
+    // TODO should we remove setExtra here, or do we need it for backwards compat?
```

Comment:
I'd keep it for now

Reply:
@stefanosiano as per the latest discussion (here)[https://github.com/getsentry/sentry-java/issues/3626#issuecomment-2324432211] I'll remove it

Reply:
@stefanosiano still in some discussions about this :) as both not setting the data in `extras` as well as adding data to `trace.context.data` might result in breaking changes for users

Reply:
Closing this for now, created a PR that targets 8.x which removes this (as both keeping and removing this would constitute a breaking change).